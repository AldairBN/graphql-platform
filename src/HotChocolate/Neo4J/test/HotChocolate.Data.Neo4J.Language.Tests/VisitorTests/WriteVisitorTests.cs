using Snapshooter.Xunit;
using Xunit;

namespace HotChocolate.Data.Neo4J.Language;

public class WriteVisitorTests
{
    private static readonly Node _movie = Node.Create("Movie")
        .Named("m")
        .WithProperties("Title", Cypher.LiteralOf("The Matrix"));

    private static readonly Pattern _pattern = new (_movie);

    [Fact]
    public void Create()
    {
        var visitor = new CypherVisitor();

        var create = new Create(_pattern);
        create.Visit(visitor);
        visitor.Print().MatchSnapshot();
    }

    [Fact]
    public void Merge()
    {
        var visitor = new CypherVisitor();

        var merge = new Merge(_pattern);
        merge.Visit(visitor);
        visitor.Print().MatchSnapshot();
    }
}
