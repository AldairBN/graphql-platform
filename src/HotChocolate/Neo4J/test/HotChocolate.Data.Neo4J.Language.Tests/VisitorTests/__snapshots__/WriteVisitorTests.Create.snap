CREATE (m:Movie  {Title: 'The Matrix'}) 
